File: Searching/PythonCodes/Leetcode350.py
class Solution(object):
    def binary_search(self, nums, target):
        left, right = 0, len(nums) - 1
        while left <= right:
            mid = left + (right - left) // 2
            if nums[mid] == target:
                return mid
            elif nums[mid] > target:
                right = mid - 1
            else:
                left = mid + 1
        return -1

    def intersect(self, nums1, nums2):
        nums1.sort()
        nums2.sort()
        result = []
        for num in nums1:
            index = self.binary_search(nums2, num)
            if index != -1:
                result.append(num)
                nums2[index] = -1  
                nums2.sort()  
        
        return result

File: Searching/JavaCodes/Leetcode350.java
class Solution {
    public static int binarySearch(int[] nums,int target){
        int start=0,end= nums.length-1;
        while(start<=end){
            int mid=start+(end-start)/2;
            if(nums[mid]==target){
                return mid;
            }
            if(nums[mid]>target){
                end=mid-1;
            }
            else{
                start=mid+1;
            }
        }
        return -1;
    }


    public static int[] intersect(int[] a, int[] b) {
        Arrays.sort(a);
        Arrays.sort(b);

        List<Integer> list=new ArrayList<>();

        for (int i = 0; i < a.length; i++) {
            int target=a[i];
            int index=binarySearch(b,target);
            if(index!=-1){
                b[index]=-1;
                list.add(target);
                Arrays.sort(b);
            }
        }

        int[] result=new int[list.size()];
        for (int i = 0; i < result.length; i++) {
            result[i]=list.get(i);
        }
        return result;
    }

}